addons/website/models/ir_http.py

```diff
@@ -90,6 +90,51 @@ def _get_default_lang(cls):
         return super(Http, cls)._get_default_lang()
 
     @classmethod
+    def _serve_page(cls, req_page=False):
+        req_page = req_page or request.httprequest.path
+
+        domain = [('url', '=', req_page), '|', ('website_ids', 'in', request.website.id), ('website_ids', '=', False)]
+
+        if not request.website.is_publisher:
+            domain += [('website_published', '=', True), '|', ('date_publish', '=', False), ('date_publish', '>', datetime.datetime.now())]
+
+        mypage = request.env['website.page'].search(domain, limit=1)
+
+        if mypage:
+            return mypage.ir_ui_view_id.render({
+                'path': req_page[1:],
+                'deletable': True,
+                'main_object': mypage,
+            })
+
+        return request.website.is_publisher() and request.render('website.page_404', {'path': req_page[1:]}) or False
```

Comment:
or False ?

In old code: handleException 404 ?

Reply:
[OK] Taken care of with JKE [12/09]